.github/workflows/test.yml

```diff
@@ -0,0 +1,37 @@
+name: Test
+
+on:
+  push:
+    branches:
+    - main
+  pull_request:
+    branches:
+    - main
+
+env:
+  PYTEST_ADDOPTS: "--color=yes"
+
+jobs:
+  build:
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ubuntu-latest, windows-latest, macos-latest]
+        python-version: ["3.8", "3.9", "3.10", "3.11", "3.12"]
```

Comment:
I see this matrix slightly suboptimal.
https://blog.jaraco.com/efficient-use-of-ci-resources